pkg/collector/corechecks/embed/apm.go

```diff
@@ -64,11 +97,30 @@ func (c *APMCheck) Run() error {
 		}
 	}()
 
-	if err = c.cmd.Start(); err != nil {
-		return err
+	if err = cmd.Start(); err != nil {
+		return retryExitError(err)
 	}
 
-	return c.cmd.Wait()
+	processDone := make(chan error)
+	go func() {
+		processDone <- cmd.Wait()
+	}()
+
+	select {
+	case err = <-processDone:
+		return retryExitError(err)
+	case <-c.stop:
+		err := cmd.Process.Signal(os.Kill)
+		if err != nil {
+			log.Errorf("unable to stop APM check: %s", err)
+		}
+		go func() {
+			// poll the processDone channel to make sure the goroutine sending to it can exit
+			<-processDone
+		}()
+		c.stopDone <- struct{}{}
```

Comment:
should this be inside the goroutine, after reading `processDone`, are we really done until that happens? Or alternatively, could we do the reading from `processDone` in the function's routine since `processDone <- cmd.Wait()` is async?

Reply:
yeah that makes sense, we can read from `processDone` in the function's goroutine, sounds like a good improvement, I'll change that 👍